Update the user-timing IDL file

lukebjerring opened the pull request:
<!-- Reviewable:start -->

<!-- Reviewable:end -->

Commit:
Add user-timing idl file

foolip commented on interfaces/user-timing.idl:
```diff
@@ -0,0 +1,17 @@
+
+        partial interface Performance {
```
Can you update to strip the indentation and use the file in https://github.com/w3c/web-platform-tests/blob/master/user-timing/idlharness.html?

foolip replied:
@lukebjerring, indentation problem still persists.

foolip commented:
Hmm, and why is Travis unhappy?

Commit:
Merge branch 'master' into idl-file-updates-user-timing

Commit:
Updated user-timing IDL file

Commit:
Updated user-timing IDL file

foolip commented:
@lukebjerring looks like this needs a test to exercise the IDL?

Commit:
Updated user-timing IDL file

Commit:
Use IDL in test + change to .any.js test

lukebjerring commented:
I've updated the `user-timing/idlharness.html` test to be `.any.js`, since it's exposed on both worker + window, and changed the test to use the IDL. PTAL.

lukebjerring commented:
cc @npm1 who was helping me work out whether using `.any.js` was the right thing to do. Note that we see a large number of `.any.worker.html` results with:

> assert_own_property: self does not have own property "PerformanceMark" expected property "PerformanceMark" missing

Goes a bit beyond my understanding of the implementation of idlharness. @foolip - any idea whether that's to be expected?

plehegar commented on user-timing/idlharness.any.js:
```diff
@@ -0,0 +1,21 @@
+promise_test(async () => {
+  const idl_array = new IdlArray();
+  const idl = await fetch("/interfaces/user-timing.idl").then(r => r.text());
+  const hrtime = await fetch("/interfaces/hr-time.idl").then(r => r.text());
```
you need to import /interfaces/performance-timeline.idl as well, since PerformanceMark depends on PerformanceEntry.

lukebjerring replied:
Alright, I'll pull in https://github.com/web-platform-tests/wpt/pull/9809 to this PR.

Commit:
Add performance-timeline idl file

Commit:
Merge branch 'master' into idl-file-updates-user-timing

Commit:
Merge branch 'idl-file-updates-performance-timeline' into idl-file-updates-user-timing

Commit:
Update performance-timeline idl test to .any.js

Commit:
Remove old idlharness test

foolip reviewed (approved):
Tested locally, .worker.html tests will reveal some things that don't match the spec in Chromium :) :+1:
